sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -219,29 +214,53 @@ def incr_mean_variance_axis0(X, last_mean, last_var, unsigned long last_n):
     `utils.extmath._batch_mean_variance_update`.
 
     """
-    cdef unsigned long n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
+    is_CSR = isinstance(X, sp.csr_matrix)
+    is_CSC = isinstance(X, sp.csc_matrix)
+    return _incr_mean_variance_axis0(X.data, X.shape, X.indices, X.indptr,
+                                     last_mean, last_var, last_n,
+                                     is_CSR, is_CSC)
```

Comment:
Ah I see why.

This might be nitpicking but you need not pass `is_CSR` or `is_CSC`. You can get whatever information you want from len(X_indptr)

    if len(X_indptr) == shape[0] + 1:
        # Then CSR
    else:
       # Then CSC